### PR TITLE
add all 3.x release logs for Darshan

### PR DESCRIPTION
**Pull request by shanedsnyder**
This PR copies over all x86-64 release logs for 3.x versions of Darshan. The plan is to maintain these release logs here going forward rather than in the darshan source repo. Using these tests in our CI will help make sure no regressions are introduced into libdarshan-util backwards compatibility code, like https://github.com/darshan-hpc/darshan/issues/762.

Eventually, these logs will be removed from the darshan repo, and we will modify our release checklist to ensure we upload updated release logs here going forward. I'll keep this PR opened until we release Darshan 3.4.0 (should be within the week), and will tack on that release log before merging.

### CODE REVIEW

**Comment by tylerjereddy** [comment]
Sounds good, the testsuite will need a few tweaks to make appropriate assumptions for the properties of those new log files, as you can see from the CI, but that should be straightforward.

**Comment by shanedsnyder** [comment]
Also need README and init.py, so I don't forget.

**Comment by tylerjereddy** [comment]
See also https://github.com/darshan-hpc/darshan/pull/767 for the shims needed for the `main` repo suite to pass with the new logs, pretty minor I'd say.

File: darshan_logs/release_logs/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+This directory contains logs generated for every 3.x version of Darshan.
+These logs are generated as part of the release process and are intended
+to ensure Darshan maintains backwards compatibility for parsing log files
+generated by older versions of the library.
+
+These test logs are generated with the mpi-io-test benchmark that is
+distributed as part of Darshan's internal regression testing framework
+(darshan-test/regression/test-cases/src/mpi-io-test.c). The benchmark
+is ran using 4 processes and uses all default parameters to mpi-io-test.
```
Comment (carns) [inline darshan_logs/release_logs/README.md:9]: This is really piddling, but "The  benchmark is ran" sounds funny to me :)  Maybe "The benchmark is executed"?

**Review by carns — CHANGES_REQUESTED** [review]
looks great, other than me being pedantic about grammar

File: darshan_logs/release_logs/README.md (hunk 1 above)
Comment (shanedsnyder) [reply darshan_logs/release_logs/README.md]: no worries, i just pushed a fix for this :)

**Review by carns — APPROVED** [review]
thanks ;)

**Pull request merged**